### PR TITLE
test: improve alert

### PR DESCRIPTION
**Pull request by DaniAcu**
Changelog
- Add `@babel/plugin-transform-runtime` to be able to use `async`/`await`
- Add `@testing-library/jest-dom/extend-expect` to be able to use the expect extentions
- Update alert test to cover all the cases
  - Prevent use classes to get the elements, instead I used the `queryByRole`. 
  - Add dismissible, controllable and without fade cases

### CODE REVIEW

**Comment by DaniAcu** [comment]
@bestguy I noticed that we are using a prop to the `toggle` event which could be `onClose`. However I wondering why to use a dispatch as svelte recommend instead to pass a function as parameters 🤔.

I'm a React fan so I like how this looks like but I'm thinking if we could do this more in a svelte way.

**Comment by bestguy** [comment]
Great, thank you @DaniAcu !  I'll take a look, that makes sense

**Comment by DaniAcu** [comment]
Let me know if the tests looks good or I need to change something. I could continue contributing with this kind of things 😄

**Comment by bestguy** [comment]
Thanks, it looks good so far, but I want to be cautious on adding babel changes incase it impacts the published library.  It should be fine, but I'll confirm over the weekend. Thank you for the help 👍

**Comment by DaniAcu** [comment]
@bestguy I noticed that the bundle increase 10KB with the babel plugin. I could configure a specific babel config for testing if that work for you 😄

**Comment by bestguy** [comment]
Yes that sounds great.  Are you sure async was not supported already in tests?

**Comment by DaniAcu** [comment]
Mmm pretty sure, it was failing because the `regeneratorRuntime` was not defined which means that the babel pollyfill was not included. So I just add the regenerator plugin that we need to prevent to add all the pollyfills

**Comment by DaniAcu** [comment]
Nice, I will continue adding other config or maybe use NODE_ENV, because we don't want to use node as target for a web component library, it should continue using web as target. Thanks for the suggestion 😊

**Comment by bestguy** [comment]
Oh yes true, in that case I think can use the babel env test blocks for this, instead of referencing NODE_ENV in code.

**Comment by DaniAcu** [comment]
@bestguy haha yeah! I pushed that at the same time that you mention haha. I have a problem with package-lock.json. Let me take a look

**Comment by DaniAcu** [comment]
Now it should be fine 😄

**Comment by DaniAcu** [comment]
If this is fine, I'll continue cleaning other tests and trying to add new ones to have a better quality in our code.

**Comment by bestguy** [comment]
Thanks @DaniAcu, I'll sync and try this evening.  I might have a suggestion on the babel config but let me confirm first 👍

**Comment by DaniAcu** [comment]
Any update? 🤔

File: babel.config.js [hunk 1]
```diff
@@ -1,3 +1,14 @@
...
+
+  if(isTest) return testingConfig;
+
+  return {
+    presets: [
+      '@babel/preset-env',
+      '@babel/preset-typescript'
+    ]
+  };
 };
```
Comment (bestguy) [inline babel.config.js:14]: Hi @DaniAcu ,  after looking at this, I think we could simplify the these babel config changes to:

```javascript
module.exports = {
  presets: ['@babel/preset-env', '@babel/preset-typescript'],
  env: {
    test: {
      presets: [[
        '@babel/preset-env',
        {
          "targets": {
            "node": "current"
          }
        }
      ], '@babel/preset-typescript'],
    }
  }
};
```

This seems to work and would avoid the additional `babel.test.config.js` file.
What do you think?

**Review by bestguy — CHANGES_REQUESTED** [review]
Thanks for your patience, one question.

File: babel.config.js (hunk 1 above)
Comment (DaniAcu) [reply babel.config.js:21]: Solved 😉

**Review by bestguy — APPROVED** [review]
Great thank you!

**Pull request merged**